command/scripting.go

```diff
@@ -0,0 +1,370 @@
+
+	// RedirectStandardErrorToOutput allows you to capture
+	// standard error in the same stream as standard output. This
+	// improves the synchronization of these streams.
+	RedirectStandardErrorToOutput bool `mapstructure:"redirect_standard_error_to_output"`
+
+	// ContinueOnError determines whether or not a failed return code
+	// should cause the task to be marked as failed. Setting this to true
+	// allows following commands to execute even if this shell command fails.
+	ContinueOnError bool `mapstructure:"continue_on_err"`
+
+	base
+}
+
+func subprocessScriptingFactory() Command {
+	return &scriptingExec{}
+}
+
+func (c *scriptingExec) Name() string { return "subprocess.scripting" }
+func (c *scriptingExec) ParseParams(params map[string]interface{}) error {
+	err := mapstructure.Decode(params, c)
+	if err != nil {
+		return errors.Wrapf(err, "error decoding %s params", c.Name())
+	}
+
+	if c.Command != "" {
+		if c.Script != "" || len(c.Args) > 0 {
+			return errors.New("must specify command as either arguments or a command string but not both")
+		}
+
+		c.Args, err = shlex.Split(c.Command)
+		if err != nil {
+			return errors.Wrapf(err, "problem parsing %s command", c.Name())
+		}
+	}
+
+	if c.Script == "" && len(c.Args) == 0 {
+		return errors.New("must specify either a script or a command")
+	}
+	if c.Script != "" && len(c.Args) > 0 {
+		return errors.New("must specify either a script or a command, but not both")
+	}
+
+	if c.CacheDurationSeconds < 1 {
+		c.CacheDurationSeconds = 10
+	}
+
+	if c.Silent {
+		c.IgnoreStandardError = true
+		c.IgnoreStandardOutput = true
+	}
+
+	if c.IgnoreStandardOutput && c.RedirectStandardErrorToOutput {
+		return errors.New("cannot ignore standard out, and redirect standard error to it")
+	}
+
+	if c.Env == nil {
+		c.Env = make(map[string]string)
+	}
+	return nil
+}
+
+func (c *scriptingExec) doExpansions(exp *util.Expansions) error {
+	var err error
+	catcher := grip.NewBasicCatcher()
+
+	c.Harness, err = exp.ExpandString(c.Harness)
+	catcher.Add(err)
+
+	c.Script, err = exp.ExpandString(c.Script)
+	catcher.Add(err)
+
+	c.WorkingDir, err = exp.ExpandString(c.WorkingDir)
+	catcher.Add(err)
+
+	c.LockFile, err = exp.ExpandString(c.LockFile)
+	catcher.Add(err)
+
+	c.HarnessPath, err = exp.ExpandString(c.HarnessPath)
+	catcher.Add(err)
+
+	for idx := range c.Packages {
+		c.Packages[idx], err = exp.ExpandString(c.Packages[idx])
+		catcher.Add(err)
+	}
+
+	for idx := range c.Args {
+		c.Args[idx], err = exp.ExpandString(c.Args[idx])
+		catcher.Add(err)
+	}
+
+	for k, v := range c.Env {
+		c.Env[k], err = exp.ExpandString(v)
+		catcher.Add(err)
+	}
+
+	if len(c.Path) > 0 {
+		path := make([]string, len(c.Path), len(c.Path)+1)
+		for idx := range c.Path {
+			path[idx], err = exp.ExpandString(c.Path[idx])
+			catcher.Add(err)
+		}
+		path = append(path, os.Getenv("PATH"))
+
+		c.Env["PATH"] = strings.Join(path, string(filepath.ListSeparator))
+	}
+
+	expansions := exp.Map()
+	if c.AddExpansionsToEnv {
```

Comment:
Do we need to document that expansions can overwrite the environment specified in the options?

Reply:
nah